### ISSUE TITLE
Z-axis position is incorrect on display

### ISSUE THREAD

**Issue by Piccyman** [johncarlson21/SV04-Marlin-2.1.x#13]
Go to move screen and the z-axis reading decimal point is out by 10
It reads 1.20 mm instead of 12.0 mm

It just a display error

**Comment by johncarlson21** [comment]
yep.. I'm working on this..

**Comment by GMaassen** [comment]
Yes, I think this is the same on the "Move" screen, but for all axes. I'm trying to set my off-set using the Ember Prototypes CXC, but I'm one decimal place off.

**Comment by johncarlson21** [comment]
I guess I don't know what you mean by this cxc.. what are you trying to do with the move screen? like just trying to find the z?

**Comment by GMaassen** [comment]
It's an external camera, developed to find off-sets. Check the website!
What I meant was, in the Settings screen of the Sovol SV04 -> Move. The numbers on this screen seem to be in cm not in mm.

**Comment by johncarlson21** [comment]
seems like an extra step, but yes the numbers are off and need to be fixed.. punching in the numbers you put them in correctly, but it shows the wrong numbers.

**Comment by johncarlson21** [comment]
I think in your situation the problem is the screen, currently in the latest build and code, I have those * 10 but the screen is looking for * 100, however this caused a problem in the screen as those inputs can only handle so many numbers and it can only be up to like a specific number, which for the Z which can go to 400+ causes the numbers in the input to show some random messed up number. so I've been working on how to fix that in the screen so we can get 2 decimal places, but for the Z this seems not to work correctly. Also X is off once it gets to like over 350 I believe when using 2 decimal places.

**Comment by DataGhost** [comment]
That's not the only thing wrong in that screen. Doing ten "0.01mm" steps yields the same resulting position as one "1mm" step on both the screen and physically. Last time I checked 10*0.01mm == 0.1mm and not 1mm. So the button should read "0.1mm". Changing the number of decimal places from 2 to 1 will convert e.g. the wrong "-6.20 mm" (really cm) to the actually correct "-62.0 mm" without showing any more places that are adjustable. 

I'm also not sure if 0.01mm is a meaningful adjustment or if it was really meant to be there. The firmware will report the current position in 0.01mm accuracy but in order to properly display that, the variable needs to be set to "long int" (4 bytes or 2 words). A normal int is 2 bytes / 1 word in DGUS and that will obviously overflow into a "random messed up number" when you're trying to represent 400.00mm as 40000, because it can't go over 32767. However, the "Incremental adjustment" function seems to only work on single words so that's absolutely terrible and I have no clue if there's a workaround possible in that software. So that would limit this to one decimal place.

I have about 30 minutes experience with the DGUS software and I absolutely hate it already. I did not build/flash the resulting LCD firmware but I only tested it in the software "simulator" itself, and I have no idea if this is going to have any adverse effects on the actual machine, which I've only got for three days now. You'll probably know what stupid side effects this could have or if it's fine and whether or not (and where) the machine firmware needs to be changed as well. That's why I'm reluctant to just make a PR right now when I don't really have time to properly test it and be sure that I didn't break anything else. Though it also seems like this repository has kind of died so idk. I can try whipping up a PR somewhere in January probably when I've managed to build and test it myself, or otherwise I guess fork this repo if there is no response by then.

**Comment by LilBub** [comment]
Hi,
First, you undertook a great challenge with this firmware and I thank you
I'm also using the Ember Prototype XY Calibration camera and it prompts the user to enter the XY position after you align the camera cross hairs with the nozzle.  Its a pretty slick means to calibrate but like others mentioned, the output on the screen is not giving good information,  The calibration asks you to raise the nozzle up 30mm.  After homing, the nozzle is already above the bed, but not reading correctly - showing 1mm when I suspect it means 10?  (X and Y read 15.10 and 12.60 - my logic is that is 151 x 126?  Is there anything we can do to assist in correcting this bug?  I'm willing to contribute to the cause.

**Comment by johncarlson21** [comment]
yeah the problem with that is I couldn't get the decimal to come out
correctly, and I'm not working on it anymore.. you are free to fork my repo
and work on it

**Comment by LilBub** [comment]
I might try that. I'll download the files you've shared and see what I can do!

**Comment by LilBub** [comment]
> yep.. I'm working on this..

@johncarlson21 I reached out the Marlin Firmware group for some help on this.  Someone was kind enough to take a lot at the repository here and provided a potential fix with the screen firmware.

Link is here - https://github.com/MarlinFirmware/Configurations/issues/1135

**Comment by johncarlson21** [comment]
So the problem I had was with the screen not really the marlin code. I can't remember anymore. But I'm no longer working on this.

**Comment by LilBub** [comment]
Right, I recall you saying that.  I was trying to reach out and help get an answer.  They provided a fix, it appears (check link!)
